crates/gen-wasmer/src/lib.rs

```diff
@@ -1078,15 +1119,13 @@ impl Generator for Wasmer {
                     /// This function returns the `{0}Data` which needs to be
                     /// passed through to `{0}::new`.
                     fn add_to_imports(
-                        store: &wasmer::Store,
-                        imports: &mut wasmer::ImportObject,
-                    ) -> std::sync::Arc<std::sync::Mutex<{0}Data>> {{
+                        store: &mut wasmer::StoreMut<'_>,
+                        imports: &mut wasmer::Imports,
+                    ) -> wasmer::FunctionEnv<{0}Data> {{
                 ",
                 name,
             ));
-            self.push_str(
-                "let state = std::sync::Arc::new(std::sync::Mutex::new(Default::default()));\n",
-            );
+            self.push_str("let env = wasmer::FunctionEnv::new(store, Default::default());\n");
```

Comment:
We are not specifying the type of the inner struct, which I think could make things a bit confusing.

Probably having something like:
```suggestion
            self.push_str(&format!("let env = wasmer::FunctionEnv::new(store, {name}Data::default());\n"));
```

Or:

```suggestion
            self.push_str(&format!("let env: {name}Data = wasmer::FunctionEnv::new(store, Default::default());\n"));
```

would make it clearer